packages/navie/src/lib/get-most-recent-messages.ts

```diff
@@ -0,0 +1,35 @@
+import { PromptInteractionEvent } from '../interaction-history';
+import type Message from '../message';
+
+const DEFAULT_HISTORICAL_MESSAGES = (() => {
+  const env = process.env.APPMAP_NAVIE_HISTORICAL_MESSAGES;
+  if (env) {
+    const value = parseInt(env, 10);
+    if (Number.isInteger(value) && value > 0) {
+      return value;
+    }
+  }
+
+  // Default one user message, one assistant message
+  return 2;
+})();
+
+export default function getMostRecentMessages(
+  messages: Message[],
+  numMessages = DEFAULT_HISTORICAL_MESSAGES
```

Comment:
I encourage you to add a UserOption for this, such as `/history=N`. This is not likely to be used by users, but does enable fine-grained control in a scripting-type scenario, and also enables us to try different strategies from the Navie UI when she's not behaving exactly as we would like.